src/services/GenerateVerifiableCredential.ts

```diff
@@ -0,0 +1,457 @@
+	
+  	const findCheck = (type: string) =>
+  		checks.find((checkCompleted: { type: string }) => checkCompleted.type === type);
+	
+  	const getCheckObject = (check: { state: any; report: { recommendation: any; breakdown: any } }) => ({
+  		object: check,
+  		state: check.state,
+  		recommendation: check.report.recommendation,
+  		breakdown: check.report.breakdown,
+  	});
+	
+		//IBV_VISUAL_REVIEW_CHECK && DOCUMENT_SCHEME_VALIDITY && PROFILE_DOCUMENT_MATCH Currently not being consumed
+  	const MANDATORY_CHECKS = {
+  		ID_DOCUMENT_AUTHENTICITY: findCheck(YOTI_CHECKS.ID_DOCUMENT_AUTHENTICITY.type) ? getCheckObject(findCheck(YOTI_CHECKS.ID_DOCUMENT_AUTHENTICITY.type)) : null,
+  		ID_DOCUMENT_FACE_MATCH: findCheck(YOTI_CHECKS.ID_DOCUMENT_FACE_MATCH.type) ? getCheckObject(findCheck(YOTI_CHECKS.ID_DOCUMENT_FACE_MATCH.type)) : null,
+  		IBV_VISUAL_REVIEW_CHECK: findCheck(YOTI_CHECKS.IBV_VISUAL_REVIEW_CHECK.type) ? getCheckObject(findCheck(YOTI_CHECKS.IBV_VISUAL_REVIEW_CHECK.type)) : null,
+  		DOCUMENT_SCHEME_VALIDITY: findCheck(YOTI_CHECKS.DOCUMENT_SCHEME_VALIDITY_CHECK.type) ? getCheckObject(findCheck(YOTI_CHECKS.DOCUMENT_SCHEME_VALIDITY_CHECK.type)) : null,
+  		PROFILE_DOCUMENT_MATCH: findCheck(YOTI_CHECKS.PROFILE_DOCUMENT_MATCH.type) ? getCheckObject(findCheck(YOTI_CHECKS.PROFILE_DOCUMENT_MATCH.type)) : null,
+  	};
+
+		this.logger.info({ message: "Yoti Mandatory Checks" }, MANDATORY_CHECKS);
+	
+  	if (Object.values(MANDATORY_CHECKS).some((check) => check?.object === undefined)) {
+  		throw new AppError(
+  			HttpCodesEnum.BAD_REQUEST,
+  			"Missing mandatory checks in Yoti completed payload",
+  			{ MANDATORY_CHECKS },
+  		);
+  	}
+	
+  	if (
+  		Object.values(MANDATORY_CHECKS).some(
+  			(check) => check?.state !== YotiSessionDocument.DONE_STATE,
+  		)
+  	) {
+  		throw new AppError(HttpCodesEnum.BAD_REQUEST, "Mandatory checks not all completed");
+  	}
+	
+  	let credentialSubject: VerifiedCredentialSubject = {};
+	
+		//Attach individuals name information to the VC payload
+  	credentialSubject = this.attachPersonName(
+  		credentialSubject,
+  		documentFields.given_names,
+  		documentFields.family_name,
+  	);
+		//Attach individuals DOB to the VC payload
+  	credentialSubject = this.attachDOB(credentialSubject, documentFields.date_of_birth);
+		const isAddressPresent = documentFields.structured_postal_address;
+		this.logger.info({ message: "Does document contain address details" }, isAddressPresent);
+		//If address info present in Media document dttach individuals Address to the VC payload
+  	if (isAddressPresent) {
+  		credentialSubject = this.attachAddressInfo(
+  			credentialSubject,
+  			documentFields.structured_postal_address,
+  	);
+  	}
+		//Attach evidence block to the VC payload
+  	credentialSubject = this.attachEvidencePayload(
+  		credentialSubject,
+  		documentType,
+			yotiCountryCode,
+  		documentFields,
+  	);
+	
+		//Assumption here that same subCheck won't be sent twice within same Check
```

Comment:
who is making that assumption? what happens if the same subCheck is sent twice?

Reply:
Assumption was that the same check won't appear twice in the completed session payload, in the case that we do .some() will check atleast one matches our desired outcome and if so will set the flah accordingly